### ISSUE TITLE
Removing the default "0" input value on init

### ISSUE THREAD

**Issue by manuelagard** [uhlryk/angular-dynamic-number#8]
Hi,

It's me again !
Do you think it would be possible to remove the default "0" value when the field in initialized ? because, as the field has already the "0", we don't see the placeholder... When using angular material, it can be a problem...
Merci ! ;-)

**Comment by uhlryk** [comment]
It is good idea!

**Comment by stevehanson** [comment]
I would love to see this as well! Just started using this today. Love it, but the zeroes are bugging me :)

**Comment by stevehanson** [comment]
I played around a bit, and it looks like changing this line: https://github.com/uhlryk/angular-dynamic-number/blob/master/src/numberDirective.js#L204 to return `null` instead of `"0"` does the trick. I haven't really tested much though.

**Comment by uhlryk** [comment]
@stevehanson thank you for your help. Your solution helps me to solve this bug, at first I used it 'as is', but I have found one little problem. 
When default model value is '' (empty string) then in function `filterModelValue` 
line: 

```
value = Number(value);
```

will return 0 instead of NaN (because Number('') === 0). And if value is 0 then this condition:

```
if(!isNaN(value) && isFinite(value)) {
```

is true and therefore function will return 0.

I made another condition to return empty value when init value is '' || null || undefined
